toolkit/tools/imagegen/configuration/parse_partition.go

```diff
@@ -89,7 +89,18 @@ func processDisk(inputDiskValue string) (err error) {
 			disks = append(disks, Disk{})
 		}
 
-		disks[latestDiskIndex].PartitionTableType = PartitionTableTypeGpt
+		// Read from /proc/cmdline to determine if the user has specified "--gpt=xxx", which
+		// indicates a GPT partitioning
+		partitionTableType, err := GetKernelCmdLineValue("--gpt")
+		if err != nil {
+			return err
+		}
+
+		if strings.TrimSpace(partitionTableType) == "True" {
```

Comment:
>strings.TrimSpace(partitionTableType) == "True" [](http://example.com/codeflow?start=5&length=47)

Seeing the possible values, I'd suggest re-naming the `partitionTableType` variable to something close to what it contains. Perhaps `isGPTPartitionTable`?

Reply:
Fixed